<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chris Simoes" />


<title>Analise Conjunta Virginia</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">AnaliseR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/chris263/AnaliseR">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analise Conjunta Virginia</h1>
<h4 class="author">Chris Simoes</h4>
<h4 class="date">3/24/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>AnaliseR/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210324code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210324)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210324code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210324)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomchris263AnaliseRtree10df00dce11a978fc0283a884dc0058fefe97bb5targetblank10df00da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/chris263/AnaliseR/tree/10df00dce11a978fc0283a884dc0058fefe97bb5" target="_blank">10df00d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomchris263AnaliseRtree10df00dce11a978fc0283a884dc0058fefe97bb5targetblank10df00da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/chris263/AnaliseR/tree/10df00dce11a978fc0283a884dc0058fefe97bb5" target="_blank">10df00d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/analiseConjuntaVA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/analiseConjuntaVA.Rmd</code>) and HTML (<code>docs/analiseConjuntaVA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chris263/AnaliseR/10df00dce11a978fc0283a884dc0058fefe97bb5/docs/analiseConjuntaVA.html" target="_blank">10df00d</a>
</td>
<td>
chris263
</td>
<td>
2021-03-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chris263/AnaliseR/blob/df218f9aa1a4bb6476950ee619cbaa5e771b4757/analysis/analiseConjuntaVA.Rmd" target="_blank">df218f9</a>
</td>
<td>
chris263
</td>
<td>
2021-03-25
</td>
<td>
Joint Analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objectives" class="section level2">
<h2>Objectives</h2>
<ol style="list-style-type: decimal">
<li>Find the best candidate to become a commercial variety</li>
<li>Develop a selection index</li>
</ol>
<div id="script-setup" class="section level3">
<h3>Script setup</h3>
<p>Install packages, start the README.md, set the random seed</p>
<pre class="r"><code>va &lt;- read.csv(&quot;data/A427_converted.csv&quot;, sep = &quot;,&quot;, header = T)
name_cols &lt;- c(&quot;germplasmName&quot;, &quot;Replication&quot;, &quot;Location&quot;, &quot;kgHa&quot;, &quot;DollarKg&quot;, &quot;DollarHa&quot;, 
    &quot;IQS&quot;, &quot;Talo&quot;, &quot;TotalSuggar&quot;, &quot;TotalAlcaloids&quot;)
colnames(va) &lt;- name_cols

Traits &lt;- colnames(va)[4:length(va)]
nTraits &lt;- ncol(va) - 3
Treatments &lt;- unique(va$germplasmName)
Locations &lt;- unique(va$Location)

#This is the economic value for each trait
econValue &lt;- c(1.5, 1.01, 1.45, 1.55, -1.25, -1.1, -1.16)


pre_result &lt;- matrix(data = NA, nrow = nTraits, ncol = 3)
colnames(pre_result) &lt;- c(&quot;h2&quot;, &quot;varG&quot;, &quot;varP&quot;)
rownames(pre_result) &lt;- Traits

library(&quot;lme4&quot;)
library(&quot;emmeans&quot;)

#Centering the data - Function scale sets mean = 0 
#It is important to scale the data in order to get a sum of different traits afterwards
va_centered &lt;- va

for (i in 1:nTraits) {
    va_centered[, 3 + i] &lt;- scale(va[, i + 3], scale = TRUE)
}

for (i in 4:length(va_centered)) {
    
    Pheno &lt;- colnames(va_centered)[i]
    model &lt;- lmer(get(Pheno) ~ (1 | germplasmName) + (1 | Replication) + (1 | Location) + 
        (1 | germplasmName:Location), na.action = na.exclude, data = va_centered)
    
    variance = as.data.frame(VarCorr(model))
    gvar &lt;- variance[2, &quot;vcov&quot;]
    envar &lt;- variance[3, &quot;vcov&quot;]
    resvar &lt;- variance[5, &quot;vcov&quot;]
    addCor &lt;- variance[2, &quot;sdcor&quot;]
    phenovar &lt;- gvar + envar + resvar
    H2 = gvar/phenovar
    pre_result[i - 3, 1] &lt;- H2
    pre_result[i - 3, 2] &lt;- gvar
    pre_result[i - 3, 3] &lt;- phenovar
    
}

preresults_table &lt;- knitr::kable(pre_result)
preresults_table</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">h2</th>
<th align="right">varG</th>
<th align="right">varP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">kgHa</td>
<td align="right">0.0708469</td>
<td align="right">0.0707958</td>
<td align="right">0.9992790</td>
</tr>
<tr class="even">
<td align="left">DollarKg</td>
<td align="right">0.2382930</td>
<td align="right">0.2211305</td>
<td align="right">0.9279773</td>
</tr>
<tr class="odd">
<td align="left">DollarHa</td>
<td align="right">0.2029126</td>
<td align="right">0.2007283</td>
<td align="right">0.9892356</td>
</tr>
<tr class="even">
<td align="left">IQS</td>
<td align="right">0.2278214</td>
<td align="right">0.2109315</td>
<td align="right">0.9258634</td>
</tr>
<tr class="odd">
<td align="left">Talo</td>
<td align="right">0.2885476</td>
<td align="right">0.2889656</td>
<td align="right">1.0014486</td>
</tr>
<tr class="even">
<td align="left">TotalSuggar</td>
<td align="right">0.0741628</td>
<td align="right">0.0771881</td>
<td align="right">1.0407916</td>
</tr>
<tr class="odd">
<td align="left">TotalAlcaloids</td>
<td align="right">0.3125941</td>
<td align="right">0.3228502</td>
<td align="right">1.0328094</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pre-processing" class="section level2">
<h2>Pre processing</h2>
<p>Preparing for Pheno and Geno matrix Calculating regression coefficients</p>
<pre class="r"><code>
##regression coefficient - if necessary
# rP &lt;- c()
# count=1
# for(i in 1:nTraits){
#   t1 &lt;- as.numeric(va[,i+3])
#   var1 &lt;- pre_result[i,3]
#   if(i &lt; nTraits){
#     for (j in (i+1):nTraits){
#       t2 &lt;- as.numeric(va[,j+3])
#       var2 &lt;- pre_result[j,3]
#       cal_rP &lt;- var(t1,t2, na.rm = TRUE)/sqrt(var1*var2)
#       rP[count] &lt;- cal_rP
#       count=count+1
#     }
#   }
# }

# 1 Phenotypic matrix
#The of diagonal is defined by the covariation between the traits.
# cov(1,2) = rP x sqrt(varPheno[1] x varPheno[2])
PhenoMatrix &lt;- matrix(data = NA, ncol = nTraits, nrow = nTraits)
for (i in 1:nTraits) {
    t1 &lt;- as.numeric(va_centered[, i + 3])
    if (i &lt; nTraits) {
        for (j in (i + 1):nTraits) {
            t2 &lt;- as.numeric(va_centered[, j + 3])
            offDiagonal &lt;- var(t1, t2, na.rm = TRUE)
            PhenoMatrix[i, j] &lt;- offDiagonal
            PhenoMatrix[j, i] &lt;- offDiagonal
        }
    }
    PhenoMatrix[i, i] &lt;- pre_result[i, 3]
}


# 2 Genotypic matrix
GenoMatrix &lt;- diag(c(pre_result[, 2]))


# 3 Economic value
# Already defined as econValue

# Calculate the Index. It is a matrix with 1 column and nTraits rows
SelectionIndex &lt;- solve(PhenoMatrix) %*% GenoMatrix %*% econValue
rownames(SelectionIndex) &lt;- Traits
colnames(SelectionIndex) &lt;- &quot;Index&quot;

print(SelectionIndex)
                    Index
kgHa           -1.0200317
DollarKg        2.0116488
DollarHa        1.1858984
IQS            -1.6080331
Talo           -0.6123607
TotalSuggar    -1.3899197
TotalAlcaloids -0.5495333</code></pre>
</div>
<div id="making-selection-with-index" class="section level2">
<h2>Making Selection with Index</h2>
<p>Here I’m using the index to multiply by each trait. After that it will be possible make a rank.</p>
<pre class="r"><code>library(dplyr)
library(tidyr)

#Calculate the index per germplasmName
preresults &lt;- va_centered %&gt;% group_by(germplasmName) %&gt;% summarise(kgHa = mean(kgHa, 
    na.rm = TRUE), DollarKg = mean(DollarKg, na.rm = TRUE), DollarHa = mean(DollarHa, 
    na.rm = TRUE), IQS = mean(IQS, na.rm = TRUE), Talo = mean(Talo, na.rm = TRUE), 
    TotalSuggar = mean(TotalSuggar, na.rm = TRUE), TotalAlcaloids = mean(TotalAlcaloids, 
        na.rm = TRUE))

#Removing extra column with germplasmName
preresults &lt;- data.matrix(subset(preresults, select = -c(germplasmName)))

#Naming rows and cols
colnames(preresults) &lt;- Traits
rownames(preresults) &lt;- Treatments

#Index Matrix has 1 column and the number of rows is the number of germplasmName
index_matrix &lt;- preresults %*% SelectionIndex

# Creating a matrix with real average (not centered)
preresults2 &lt;- va %&gt;% group_by(germplasmName) %&gt;% summarise(kgHa = mean(kgHa, na.rm = TRUE), 
    DollarKg = mean(DollarKg, na.rm = TRUE), DollarHa = mean(DollarHa, na.rm = TRUE), 
    IQS = mean(IQS, na.rm = TRUE), Talo = mean(Talo, na.rm = TRUE), TotalSuggar = mean(TotalSuggar, 
        na.rm = TRUE), TotalAlcaloids = mean(TotalAlcaloids, na.rm = TRUE))

preresults2 &lt;- data.matrix(subset(preresults2, select = -c(germplasmName)))

#print(index_matrix)
# Result matrix has all traits + selection index
results &lt;- as.data.frame(cbind(preresults2, index_matrix))
results &lt;- results[order(results$Index, decreasing = TRUE), ]

#Print the result with Rmarkdown 
results_table &lt;- knitr::kable(results)
results_table</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">kgHa</th>
<th align="right">DollarKg</th>
<th align="right">DollarHa</th>
<th align="right">IQS</th>
<th align="right">Talo</th>
<th align="right">TotalSuggar</th>
<th align="right">TotalAlcaloids</th>
<th align="right">Index</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="right">4006.970</td>
<td align="right">3.064545</td>
<td align="right">12239.588</td>
<td align="right">67.01515</td>
<td align="right">28.86515</td>
<td align="right">12.87576</td>
<td align="right">3.286667</td>
<td align="right">0.7989638</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="right">4061.879</td>
<td align="right">3.314545</td>
<td align="right">13469.310</td>
<td align="right">73.61515</td>
<td align="right">30.11636</td>
<td align="right">14.92727</td>
<td align="right">3.359091</td>
<td align="right">0.4829939</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="right">4154.939</td>
<td align="right">3.260606</td>
<td align="right">13545.349</td>
<td align="right">71.46667</td>
<td align="right">30.43152</td>
<td align="right">14.67879</td>
<td align="right">3.311818</td>
<td align="right">0.4589203</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">4075.606</td>
<td align="right">3.180303</td>
<td align="right">12990.008</td>
<td align="right">68.75152</td>
<td align="right">29.09000</td>
<td align="right">14.09697</td>
<td align="right">3.780909</td>
<td align="right">0.4484802</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="right">3912.344</td>
<td align="right">3.195625</td>
<td align="right">12466.723</td>
<td align="right">72.58437</td>
<td align="right">30.98194</td>
<td align="right">13.12903</td>
<td align="right">3.296129</td>
<td align="right">0.2334193</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="right">3904.485</td>
<td align="right">2.908485</td>
<td align="right">11373.517</td>
<td align="right">64.38788</td>
<td align="right">30.89545</td>
<td align="right">12.35455</td>
<td align="right">3.033030</td>
<td align="right">0.0088055</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="right">4109.091</td>
<td align="right">2.938485</td>
<td align="right">12055.652</td>
<td align="right">63.18788</td>
<td align="right">31.42061</td>
<td align="right">12.67576</td>
<td align="right">3.215152</td>
<td align="right">-0.0787915</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="right">3606.606</td>
<td align="right">2.613939</td>
<td align="right">9496.347</td>
<td align="right">55.07879</td>
<td align="right">31.29970</td>
<td align="right">10.26970</td>
<td align="right">3.378788</td>
<td align="right">-0.1506069</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="right">4060.333</td>
<td align="right">3.073939</td>
<td align="right">12507.655</td>
<td align="right">66.46667</td>
<td align="right">32.50303</td>
<td align="right">12.43939</td>
<td align="right">3.790000</td>
<td align="right">-0.2862362</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="right">3970.758</td>
<td align="right">3.038182</td>
<td align="right">12038.190</td>
<td align="right">66.76667</td>
<td align="right">30.71909</td>
<td align="right">13.02121</td>
<td align="right">3.755758</td>
<td align="right">-0.3246390</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="right">4020.273</td>
<td align="right">3.157879</td>
<td align="right">12701.920</td>
<td align="right">69.01212</td>
<td align="right">32.31121</td>
<td align="right">13.01212</td>
<td align="right">3.919394</td>
<td align="right">-0.3482980</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="right">4154.182</td>
<td align="right">3.143333</td>
<td align="right">13064.612</td>
<td align="right">69.04242</td>
<td align="right">32.43212</td>
<td align="right">12.41818</td>
<td align="right">3.991515</td>
<td align="right">-0.3890375</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="right">4254.242</td>
<td align="right">3.156667</td>
<td align="right">13419.774</td>
<td align="right">68.10909</td>
<td align="right">32.28091</td>
<td align="right">13.25758</td>
<td align="right">3.980909</td>
<td align="right">-0.4142106</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="right">4108.455</td>
<td align="right">3.144242</td>
<td align="right">12911.744</td>
<td align="right">68.28182</td>
<td align="right">33.16606</td>
<td align="right">12.51515</td>
<td align="right">3.941212</td>
<td align="right">-0.4247745</td>
</tr>
</tbody>
</table>
</div>
<div id="stability-analysis-with-ammi" class="section level2">
<h2>Stability analysis with AMMI</h2>
<p>This is using Bilinear sowftware</p>
<pre class="r"><code>
library(&quot;Bilinear&quot;)

# # Prepare a matrix with mean of treatments per location
preStability &lt;- data.matrix(va %&gt;% group_by(Location, germplasmName) %&gt;% summarise(kgHa = mean(kgHa, 
    na.rm = TRUE), DollarKg = mean(DollarKg, na.rm = TRUE), DollarHa = mean(DollarHa, 
    na.rm = TRUE), IQS = mean(IQS, na.rm = TRUE), Talo = mean(Talo, na.rm = TRUE), 
    TotalSuggar = mean(TotalSuggar, na.rm = TRUE), TotalAlcaloids = mean(TotalAlcaloids, 
        na.rm = TRUE)))

Stability &lt;- preStability[, 3:9] %*% SelectionIndex
Stability &lt;- cbind(preStability[, 2], preStability[, 1], Stability)
colnames(Stability) &lt;- c(&quot;germplasmName&quot;, &quot;Location&quot;, &quot;Index&quot;)
rownames(Stability) &lt;- preStability[, 2]

Stability &lt;- tapply(Stability[, 3], list(Stability[, 1], Stability[, 2]), mean)
AMMI_Index &lt;- bilinear(x = Stability, verbose = F)

#Stability using a specific trait
Trt &lt;- tapply(va$DollarKg, list(va$germplasmName, va$Location), mean)
AMMI_Trt &lt;- bilinear(x = Trt, verbose = F)
AMMIplot(AMMI_Trt)
# abline(v=-0.05, col=&#39;red&#39;)
# abline(v=+0.05, col=&#39;red&#39;)
# AMMIplot(AMMIfit, PC=2)

library(&quot;agricolae&quot;)
## Exemple: mmodel&lt;- with(va,AMMI(Location, germplasmName, Replication, DollarHa, console=FALSE))
Stability_AMMI &lt;- data.matrix(va_centered[, 4:10]) %*% SelectionIndex
va_centered &lt;- cbind(va_centered, Stability_AMMI)
colnames(va_centered) &lt;- c(name_cols, &quot;Index&quot;)

model &lt;- with(va_centered, AMMI(Location, germplasmName, Replication, Index, console = TRUE))
model$ANOVA
# # see help(plot.AMMI)
# # biplot
plot(model)
AMMI.contour(model, distance = 0.398, shape = 8, col = &quot;red&quot;, lwd = 2, lty = 5)
# # triplot PC 1,2,3 
# plot(model, type=2, number=TRUE)
# # biplot PC1 vs Yield 
# plot(model, first=0,second=1, number=TRUE)
# # Example 2

Analysis of Variance Table
 Response: y 
     Df          SS          MS   testStat       Pvalue    
E    10 346841080.9 34684108.09   38.60015 3.352979e-34 ***
G    13 172685742.1 13283518.62   14.78331 6.282633e-20 ***
PC1  22  52839538.1  2401797.19    0.00000      &lt; 1e-04 ***
PC2  20  14602248.3   730112.41 9967.00000 9.967000e-01    
PC3  18  13594573.4   755254.08 9509.00000 9.509000e-01    
PC4  16  11398379.8   712398.74 8652.00000 8.652000e-01    
PC5  14   9087132.5   649080.89 7380.00000 7.380000e-01    
PC6  12   6493252.4   541104.37 7040.00000 7.040000e-01    
PC7  10   5130129.5   513012.95 2734.00000 2.734000e-01    
PC8   8   2843701.3   355462.66  832.00000 8.320000e-02    
PC9   6    657253.0   109542.17 4045.00000 4.045000e-01    
PC10  4    165103.1    41275.78         NA                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05
Number of significant multiplicative terms  : 1 
Analysis of Variance Table
 Response: y 
     Df         SS          MS   testStat       Pvalue    
E    10 8.61503463 0.861503463   32.26864 1.376723e-30 ***
G    13 4.35352670 0.334886669   12.54358 2.043459e-17 ***
PC1  22 1.88437793 0.085653542    0.00000      &lt; 1e-04 ***
PC2  20 0.49336774 0.024668387 4806.00000 4.806000e-01    
PC3  18 0.37374472 0.020763595 4373.00000 4.373000e-01    
PC4  16 0.23304299 0.014565187 8340.00000 8.340000e-01    
PC5  14 0.19932353 0.014237395 4875.00000 4.875000e-01    
PC6  12 0.13221745 0.011018121 4900.00000 4.900000e-01    
PC7  10 0.07587707 0.007587707 6728.00000 6.728000e-01    
PC8   8 0.04475559 0.005594449 7034.00000 7.034000e-01    
PC9   6 0.02880638 0.004801064 2620.00000 2.620000e-01    
PC10  4 0.00520753 0.001301882         NA                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05
Number of significant multiplicative terms  : 1 

ANALYSIS AMMI:  Index 
Class level information

ENV:  E1 E2 E3 E4 E5 E6 E8 E10 E11 E15 E16
GEN:  1 2 3 4 5 6 7 8 9 10 11 12 13 14
REP:  1 2 3

Number of observations:  460 

model Y: Index  ~ ENV + REP%in%ENV + GEN + ENV:GEN
Random effect REP%in%ENV

Analysis of Variance Table

Response: Y
           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
ENV        10 122.271 12.2271 18.7395 1.568e-08 ***
REP(ENV)   22  14.354  0.6525  1.1665    0.2773    
GEN        13  72.536  5.5797  9.9755 &lt; 2.2e-16 ***
ENV:GEN   130 142.228  1.0941  1.9560 1.697e-06 ***
Residuals 284 158.852  0.5593                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Coeff var   Mean Index 
17608.63     0.004247284 

Analysis
     percent  acum Df    Sum.Sq  Mean.Sq F.value   Pr.F
PC1     42.3  42.3 22 60.147939 2.733997    4.89 0.0000
PC2     18.5  60.8 20 26.328474 1.316424    2.35 0.0012
PC3     14.9  75.7 18 21.148760 1.174931    2.10 0.0061
PC4      8.6  84.2 16 12.180547 0.761284    1.36 0.1609
PC5      5.8  90.0 14  8.232009 0.588001    1.05 0.4037
PC6      4.1  94.2 12  5.859328 0.488277    0.87 0.5781
PC7      3.7  97.9 10  5.311448 0.531145    0.95 0.4877
PC8      1.4  99.3  8  1.986085 0.248261    0.44 0.8964
PC9      0.6  99.9  6  0.854278 0.142380    0.25 0.9591
PC10     0.1 100.0  4  0.165209 0.041302    0.07 0.9910
Analysis of Variance Table

Response: Y
           Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
ENV        10 122.271 12.2271 18.7395 1.568e-08 ***
REP(ENV)   22  14.354  0.6525  1.1665    0.2773    
GEN        13  72.536  5.5797  9.9755 &lt; 2.2e-16 ***
ENV:GEN   130 142.228  1.0941  1.9560 1.697e-06 ***
Residuals 284 158.852  0.5593                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Limit, radio: 0.5535593
Genotype  in: 7
Genotype out: 7 

$`GENOTYPE IN`
[1] &quot;1&quot;  &quot;11&quot; &quot;13&quot; &quot;3&quot;  &quot;4&quot;  &quot;5&quot;  &quot;7&quot; 

$`GENOTYPE OUT`
[1] &quot;10&quot; &quot;12&quot; &quot;14&quot; &quot;2&quot;  &quot;6&quot;  &quot;8&quot;  &quot;9&quot; 

$Distance
     distance
1  0.13001093
10 1.19777180
11 0.35730955
12 1.39085250
13 0.23691562
14 0.64931204
2  0.60670857
3  0.49362653
4  0.03788082
5  0.48967732
6  0.64500299
7  0.16301523
8  0.91708731
9  1.14544084</code></pre>
<p><img src="figure/analiseConjuntaVA.Rmd/stability-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stability-1">
Past versions of stability-1.png
</button>
</p>
<div id="fig-stability-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/chris263/AnaliseR/blob/10df00dce11a978fc0283a884dc0058fefe97bb5/docs/figure/analiseConjuntaVA.Rmd/stability-1.png" target="_blank">10df00d</a>
</td>
<td>
chris263
</td>
<td>
2021-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/analiseConjuntaVA.Rmd/stability-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Note that the <code>echo = FALSE</code> parameter was added to the code chunk to prevent printing of the R code that generated the plot.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] agricolae_1.3-3 Bilinear_0.2.2  tidyr_1.1.2     dplyr_1.0.2    
[5] emmeans_1.5.5-1 lme4_1.1-26     Matrix_1.3-2   

loaded via a namespace (and not attached):
 [1] statmod_1.4.35   tidyselect_1.1.0 xfun_0.20        purrr_0.3.4     
 [5] haven_2.3.1      splines_4.0.3    lattice_0.20-41  labelled_2.7.0  
 [9] vctrs_0.3.6      generics_0.1.0   miniUI_0.1.1.1   htmltools_0.5.1 
[13] yaml_2.2.1       AlgDesign_1.2.0  rlang_0.4.10     later_1.1.0.1   
[17] pillar_1.4.7     nloptr_1.2.2.2   glue_1.4.2       questionr_0.7.4 
[21] lifecycle_0.2.0  stringr_1.4.0    combinat_0.0-8   workflowr_1.6.2 
[25] mvtnorm_1.1-1    coda_0.19-4      evaluate_0.14    forcats_0.5.0   
[29] knitr_1.30       fastmap_1.0.1    httpuv_1.5.4     highr_0.8       
[33] Rcpp_1.0.6       xtable_1.8-4     readr_1.4.0      promises_1.1.1  
[37] formatR_1.7      mime_0.9         fs_1.5.0         klaR_0.6-15     
[41] hms_0.5.3        digest_0.6.27    stringi_1.5.3    shiny_1.5.0     
[45] grid_4.0.3       rprojroot_2.0.2  here_1.0.1       tools_4.0.3     
[49] magrittr_2.0.1   tibble_3.0.4     cluster_2.1.0    crayon_1.3.4    
[53] whisker_0.4      pkgconfig_2.0.3  ellipsis_0.3.1   MASS_7.3-53     
[57] estimability_1.3 minqa_1.2.4      rmarkdown_2.6    rstudioapi_0.13 
[61] R6_2.5.0         boot_1.3-25      nlme_3.1-151     git2r_0.28.0    
[65] compiler_4.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
